libs/notebook/detect.go

```diff
@@ -139,3 +139,46 @@ func Detect(name string) (notebook bool, language workspace.Language, err error)
 	b := filepath.Base(name)
 	return DetectWithFS(os.DirFS(d), b)
 }
+
+type inMemoryFile struct {
+	content   []byte
+	readIndex int64
```

Comment:
FYI, you can use https://pkg.go.dev/bytes#NewBuffer and avoid the hand-rolled `Read`.

Reply:
Done